contracts/tokens/STFactory.sol

```diff
@@ -45,6 +46,7 @@ contract STFactory is ISTFactory {
         );
         //NB When dataStore is generated, the security token address is automatically set via the constructor in DataStoreProxy.
         newSecurityToken.changeDataStore(dataStoreFactory.generateDataStore(address(newSecurityToken)));
+        newSecurityToken.changeTreasuryWallet(_treasuryWallet);
```

Comment:
What's the reason to not add this to the token constructor directly?

Reply:
because datastore doesn't aware of the security token address at that time.